### ISSUE TITLE
Move-DbaDbFile failed when logical file name start with a space character

### ISSUE THREAD

**Issue by rferraton** [dataplat/dbatools#8675]
### Verified issue does not already exist?

I have searched and found no existing issue

### What error did you receive?

[Move-DbaDbFile] ERROR: | MODIFY FILE failed. File 'XXXXXX_Data' does not exist.

### Steps to Reproduce

You need to have a logical filename starting with a space
```sql
USE [DBATOOLS]
GO
ALTER DATABASE [DBATOOLS] MODIFY FILE (NAME=N'DBATOOLS', NEWNAME=N' DBATOOLS')
GO
```

```powershell
Move-DbaDbFile -SqlInstance localhost -Database DBATOOLS -FileType Data -FileDestination "C:\MSSQL\Data" -Force 
```

### Please confirm that you are running the most recent version of dbatools

1.1.143

### What PowerShell host was used when producing this error

Windows PowerShell (powershell.exe)

### PowerShell Host Version

Name                           Value                                                                                   
----                           -----                                                                                   
PSVersion                      5.1.14393.5127                                                                          
PSEdition                      Desktop                                                                                 
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                 
BuildVersion                   10.0.14393.5127                                                                         
CLRVersion                     4.0.30319.42000                                                                         
WSManStackVersion              3.0                                                                                     
PSRemotingProtocolVersion      2.3                                                                                     
SerializationVersion           1.1.0.1                                                                                 

### SQL Server Edition and Build number

Microsoft SQL Server 2016 (SP3) (KB5003279) - 13.0.6300.2 (X64)   Aug  7 2021 01:20:37   Copyright (c) Microsoft Corporation  Enterprise Edition: Core-based Licensing (64-bit) on Windows Server 2016 Standard 10.0 <X64> (Build 14393: ) 

### .NET Framework Version

PSChildName                      Version
-----------                      -------
v2.0.50727                       2.0.50727.4927
v3.0                             3.0.30729.4926
Windows Communication Foundation 3.0.4506.4926
Windows Presentation Foundation  3.0.6920.4902
v3.5                             3.5.30729.4926
Client                           4.7.03062
Full                             4.7.03062
Client                           4.0.0.0

**Comment by rferraton** [comment]
I found 2 **.trim()** in the **Get-DbaDbFile.ps1** (line 250 and 251). I suggest **TrimEnd()** instead.

```powershell
            [PSCustomObject]@{
                ComputerName             = $server.ComputerName
                InstanceName             = $server.ServiceName
                SqlInstance              = $server.DomainInstanceName
                Database                 = $db.name
                DatabaseID               = $db.ID
                FileGroupName            = $result.FileGroupName
                ID                       = $result.ID
                Type                     = $result.Type
                TypeDescription          = $result.TypeDescription
                LogicalName              = $result.LogicalName.Trim()
                PhysicalName             = $result.PhysicalName.Trim()
                State                    = $result.State
```

**Comment by wsmelton** [comment]
Marking this as a feature request. 

The module as a whole is meant to support best practices with SQL Server and PowerShell. SQL Server supporting a prefix space in the file name while possible is not a practice that is common, nor should it be.

A number of 👍 will be needed to show support that this feature is desired by the community. If not, we will eventually close this as [will not support].

**Comment by rferraton** [comment]
I totally agree that this kind of naming is a bad practice.

May I suggest to add a database check in dbachecks to validate best practises around filenames to found theses bad naming.

On the dbatools side i found quite dangerous the fact that the logical name is twister (trimed).

**Comment by wsmelton** [comment]
Feature suggestions for dbachecks are welcomed. Please submit those here: [dbachecks/issues](https://github.com/dataplat/dbachecks)

**Comment by wsmelton** [comment]
Fix will be in 2.0 release.
